status/controller_test.go

```diff
@@ -20,23 +20,20 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
 var ctx context.Context
 var recorder *record.FakeRecorder
 var kubeClient client.Client
 
-var _ = BeforeEach(func() {
-	recorder = record.NewFakeRecorder(10)
-	kubeClient = fake.NewClientBuilder().WithScheme(scheme.Scheme).Build()
-	ctx = log.IntoContext(context.Background(), ginkgo.GinkgoLogr)
-
-	status.ConditionDuration.Reset()
-	status.ConditionCount.Reset()
-	status.ConditionCurrentStatusSeconds.Reset()
-	status.ConditionTransitionsTotal.Reset()
-	status.TerminationCurrentTimeSeconds.Reset()
-	status.TerminationDuration.Reset()
+var _ = AfterEach(func() {
+	status.ConditionDuration.Unregister(metrics.Registry)
```

Comment:
You don't need to expose this as a method on the interface to get this to work. You can achieve the same by type-casting this into a prometheus metric like `status.ConditionDuration.(*metrics.PrometheusHistogram).Unregister(metrics.Registry)`

Reply:
Oh wow, that will simplify this a lot, thanks!